modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductMedia.testcase

```diff
@@ -320,6 +320,226 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12773 | Verify that it is possible to use the toggle Shopping Experience Image Gallery to decide which image shows on the storefront product details"
+	@priority = 4
+	test CanUseTheShoppingExperienceImageGalleryToggle {
+		property custom.properties = "jsonws.web.service.paths.excludes=";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium Site") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And an Account with the buyer") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And create a new product with an option linked to it") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Minium",
+				productName = "Simple_Product",
+				productType = "simple",
+				sku = "TESTSKU");
+
+			CommerceJSONOptionsAPI._assignOptionToCommerceProduct(
+				fieldType = "select",
+				optionKey = "package-quantity",
+				optionName = "Package Quantity",
+				productName = "Simple_Product");
+		}
+
+		task ("And open the product admin details") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "Simple_Product");
+		}
+
+		task ("And all SKUs are generated") {
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceSKUs.generateAllSkuCombinations();
+		}
+
+		task ("And disable the toggle Shopping Experience Image Gallery for an image") {
+			CommerceEntry.gotoMenuTab(menuTab = "Media");
+
+			CommerceProductImages.addProductImages(
+				imageTitles = "Black",
+				optionValues = 6,
+				shoppingExperienceImageGallery = "false");
+
+			CommerceProductImages.addProductImages(
+				imageTitles = "White,Blue",
+				optionValues = "12,24",
+				shoppingExperienceImageGallery = "true");
+		}
+
+		task ("And the buyer logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("When the buyer opens the product details") {
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Simple_Product");
+		}
+
+		task ("Then only two images are present into the gallery") {
+			for (var optionValues : list "6,12,24") {
```

Comment:
Within this for loop, I don't know why if the Black is the only one that is disabled, then it cannot be seen White or Blue.

Reply:
It depends on the option that I select. The image is linked on a total of 3 option values (6,12,24) when I use the select field and change the values I expect that the main image will change, I used the AssertElementNotPresent to verify that only the expected image is visible for the correct option value